### PR TITLE
[full-ci] Events

### PR DESCRIPTION
**Pull request by kobergj**
## Description
Adds a nats server to ocis and configures events
For now only a `ShareCreated` event is emitted, but no consumers are configured

## Related Issue

- Implements [events](https://jira.owncloud.com/browse/OCIS-2320)
- also bumps reva to latest edge

## Motivation and Context
Events are cool

## How Has This Been Tested?

- local testing

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: storage/pkg/command/storageusers.go [hunk 1]
```diff
@@ -129,11 +132,9 @@ func storageUsersConfigFromStruct(c *cli.Context, cfg *config.Config) map[string
 			},
 		},
 	}
-	if cfg.Reva.StorageUsers.ReadOnly {
+	if cfg.Reva.StorageShares.ReadOnly {
```
Comment (kobergj) [inline storage/pkg/command/storageusers.go:135]: Revert!

File: nats/pkg/config/log.go [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package config
+
+// Log defines the available log configuration.
+type Log struct {
+	Level  string `mapstructure:"level" env:"OCIS_LOG_LEVEL;SETTINGS_LOG_LEVEL"`
+	Pretty bool   `mapstructure:"pretty" env:"OCIS_LOG_PRETTY;SETTINGS_LOG_PRETTY"`
+	Color  bool   `mapstructure:"color" env:"OCIS_LOG_COLOR;SETTINGS_LOG_COLOR"`
+	File   string `mapstructure:"file" env:"OCIS_LOG_FILE;SETTINGS_LOG_FILE"`
```
Comment (butonic) [inline nats/pkg/config/log.go:8]: SETTINGS_ -> NATS_ or EVENTS_?

File: nats/pkg/config/http.go [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package config
+
+// HTTP defines the available http configuration.
+type HTTP struct {
+	Addr      string `ocisConfig:"addr" env:"SETTINGS_HTTP_ADDR"`
+	Namespace string
+	Root      string `ocisConfig:"root" env:"SETTINGS_HTTP_ROOT"`
+	CacheTTL  int    `ocisConfig:"cache_ttl" env:"SETTINGS_CACHE_TTL"`
```
Comment (butonic) [inline nats/pkg/config/http.go:8]: SETTINGS_ -> NATS_ or EVENTS_?

File: nats/pkg/config/http.go [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+	CORS      CORS   `ocisConfig:"cors"`
```
Comment (butonic) [inline nats/pkg/config/http.go:9]: does nats even have an http endpoint?

File: nats/pkg/config/grpc.go [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package config
+
+// GRPC defines the available grpc configuration.
+type GRPC struct {
+	Addr      string `ocisConfig:"addr" env:"SETTINGS_GRPC_ADDR"`
```
Comment (butonic) [inline nats/pkg/config/grpc.go:5]: SETTINGS_ -> NATS_ or EVENTS_?

File: nats/pkg/config/grpc.go [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package config
+
+// GRPC defines the available grpc configuration.
+type GRPC struct {
```
Comment (butonic) [inline nats/pkg/config/grpc.go:4]: does nats even have a grpc endpoint?

File: nats/pkg/config/http.go [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+}
+
+// CORS defines the available cors configuration.
+type CORS struct {
```
Comment (butonic) [inline nats/pkg/config/http.go:13]: unneded?

File: nats/pkg/config/reva.go [hunk 8]
```diff
@@ -0,0 +1,6 @@
+package config
+
+// TokenManager is the config for using the reva token manager
+type TokenManager struct {
+	JWTSecret string `ocisConfig:"jwt_secret" env:"OCIS_JWT_SECRET;SETTINGS_JWT_SECRET"`
```
Comment (butonic) [inline nats/pkg/config/reva.go:5]: SETTINGS_ -> NATS_ or EVENTS_?

File: nats/pkg/config/reva.go [hunk 9]
```diff
@@ -0,0 +1,6 @@
+package config
+
+// TokenManager is the config for using the reva token manager
+type TokenManager struct {
```
Comment (butonic) [inline nats/pkg/config/reva.go:4]: does nats need to make grpc calls? does it need the token?

File: nats/pkg/config/tracing.go [hunk 10]
```diff
@@ -0,0 +1,9 @@
+package config
+
+// Tracing defines the available tracing configuration.
+type Tracing struct {
+	Enabled   bool   `ocisConfig:"enabled" env:"OCIS_TRACING_ENABLED;SETTINGS_TRACING_ENABLED"`
+	Type      string `ocisConfig:"type" env:"OCIS_TRACING_TYPE;SETTINGS_TRACING_TYPE"`
+	Endpoint  string `ocisConfig:"endpoint" env:"OCIS_TRACING_ENDPOINT;SETTINGS_TRACING_ENDPOINT"`
+	Collector string `ocisConfig:"collector" env:"OCIS_TRACING_COLLECTOR;SETTINGS_TRACING_COLLECTOR"`
```
Comment (butonic) [inline nats/pkg/config/tracing.go:8]: SETTINGS_ -> NATS_ or EVENTS_?

File: nats/pkg/config/config.go [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+	Log     *Log     `ocisConfig:"log"`
+	Debug   Debug    `ocisConfig:"debug"`
+
+	Nats Nats `ociConfig:"nats"`
+
+	HTTP HTTP `ocisConfig:"http"`
+	GRPC GRPC `ocisConfig:"grpc"`
+
+	DataPath     string       `ocisConfig:"data_path" env:"SETTINGS_DATA_PATH"`
+	TokenManager TokenManager `ocisConfig:"token_manager"`
```
Comment (butonic) [inline nats/pkg/config/config.go:25]: unneeded?

**Review by butonic — CHANGES_REQUESTED** [review]
some c'n'p leftovers

File: nats/pkg/config/tracing.go [hunk 12]
```diff
@@ -0,0 +1,9 @@
+package config
+
+// Tracing defines the available tracing configuration.
+type Tracing struct {
```
Comment (butonic) [inline nats/pkg/config/tracing.go]: nats does not need tracing

**Review by butonic — APPROVED** [review]

**Review by micbar — APPROVED** [review]

File: nats/pkg/config/defaultconfig.go [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+// TODO: Clean up unneeded configuration
+
+func DefaultConfig() *Config {
+	return &Config{
+		Service: Service{
+			Name: "nats",
+		},
+		Nats: Nats{
+			Host: "127.0.0.1",
+			Port: 4222,
```
Comment (rhafer) [inline nats/pkg/config/defaultconfig.go:13]: @kobergj I know this is the NATS default port. But it seems we have a policy to only listen on a specific port range. (See https://github.com/owncloud/ocis/wiki/Official-Port-Ranges). Yeah, I also know I am a bit late to the party for this PR but I also just learned about that document ;-)

**Pull request merged**